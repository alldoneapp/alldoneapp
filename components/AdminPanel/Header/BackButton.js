import React from 'react'
import { StyleSheet, View } from 'react-native'

import Icon from '../../Icon'
import { colors } from '../../styles/global'
import { TouchableOpacity } from 'react-native-gesture-handler'
import NavigationService from '../../../utils/NavigationService'
import { hideFloatPopup, navigateToAllProjectsTasks, resetFloatPopup } from '../../../redux/actions'
import store from '../../../redux/store'
import { MIN_URLS_IN_HISTORY } from '../../../URLSystem/URLTrigger'
import SharedHelper from '../../../utils/SharedHelper'

export default function BackButton() {
    const onPress = () => {
        let { lastVisitedScreen } = store.getState()

        store.dispatch(hideFloatPopup())
        if (lastVisitedScreen.length >= MIN_URLS_IN_HISTORY) {
            SharedHelper.onHistoryPop('admin')
        } else {
            NavigationService.navigate('Root')
            store.dispatch([resetFloatPopup(), navigateToAllProjectsTasks()])
        }
    }

    return (
        <TouchableOpacity style={localStyles.touchableContainer} onPress={onPress}>
            <View style={localStyles.upperContainer} />
            <View style={localStyles.bottomContainer}>
                <Icon name="arrow-left" size={24} color={colors.Text03} />
            </View>
        </TouchableOpacity>
    )
}

const localStyles = StyleSheet.create({
    upperContainer: {
        width: 57,
        height: 32,
        backgroundColor: 'white',
    },
    bottomContainer: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        width: 57,
        height: 32,
        backgroundColor: 'white',
        borderRightWidth: 1,
        borderRightColor: colors.Gray300,
    },
    touchableContainer: {
        width: 57,
        height: 64,
    },
})
