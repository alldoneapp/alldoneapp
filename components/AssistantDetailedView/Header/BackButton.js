import React from 'react'
import { StyleSheet, TouchableOpacity } from 'react-native'
import { useSelector, useDispatch } from 'react-redux'

import Icon from '../../Icon'
import { SIDEBAR_MENU_COLLAPSED_WIDTH, colors } from '../../styles/global'
import NavigationService from '../../../utils/NavigationService'
import { hideFloatPopup, navigateToAllProjectsTasks, resetFloatPopup } from '../../../redux/actions'
import { MIN_URLS_IN_HISTORY } from '../../../URLSystem/URLTrigger'
import SharedHelper from '../../../utils/SharedHelper'

export default function BackButton({ assistant }) {
    const dispatch = useDispatch()
    const lastVisitedScreen = useSelector(state => state.lastVisitedScreen)
    const isMiddleScreen = useSelector(state => state.isMiddleScreen)
    const sidebarExpanded = useSelector(state => state.loggedUser.sidebarExpanded)

    const onPress = () => {
        dispatch(hideFloatPopup())
        if (lastVisitedScreen.length >= MIN_URLS_IN_HISTORY) {
            SharedHelper.onHistoryPop(`admin/assistants/${assistant.uid}`)
        } else {
            NavigationService.navigate('Root')
            dispatch([resetFloatPopup(), navigateToAllProjectsTasks()])
        }
    }

    return (
        <TouchableOpacity
            style={[
                localStyles.container,
                isMiddleScreen ? localStyles.mobileContainer : localStyles.desktopContainer,
                !isMiddleScreen && !sidebarExpanded && { marginLeft: SIDEBAR_MENU_COLLAPSED_WIDTH },
            ]}
            onPress={onPress}
        >
            <Icon name="arrow-left" size={24} color={colors.Text03} />
        </TouchableOpacity>
    )
}

const localStyles = StyleSheet.create({
    container: {
        marginTop: 32,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        width: 57,
        height: 32,
        backgroundColor: 'white',
        borderRightWidth: 1,
        borderRightColor: colors.Gray300,
    },
    mobileContainer: {
        left: -16,
    },
    desktopContainer: {
        position: 'absolute',
        top: 0,
        left: 32,
        zIndex: 100,
    },
})
