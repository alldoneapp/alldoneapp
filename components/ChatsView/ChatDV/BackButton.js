import React from 'react'
import { StyleSheet, View, TouchableOpacity } from 'react-native'
import Icon from '../../Icon'
import { colors } from '../../styles/global'
import NavigationService from '../../../utils/NavigationService'
import { setSelectedSidebarTab } from '../../../redux/actions'
import { DV_TAB_CHAT_BOARD, DV_TAB_ROOT_CHATS, DV_TAB_ROOT_NOTES } from '../../../utils/TabNavigationConstants'
import { useDispatch, useSelector } from 'react-redux'

export default function BackButton({ isFullscreen }) {
    const dispatch = useDispatch()
    const lastVisitedScreen = useSelector(state => state.lastVisitedScreen)
    const selectedTab = useSelector(state => state.selectedNavItem)

    const onPress = () => {
        NavigationService.navigate('Root')
        dispatch(
            setSelectedSidebarTab(
                lastVisitedScreen[lastVisitedScreen.length - 2].indexOf('notes') !== -1
                    ? DV_TAB_ROOT_NOTES
                    : DV_TAB_ROOT_CHATS
            )
        )
    }

    return (
        <TouchableOpacity
            style={[
                localStyles.touchableContainer,
                { top: isFullscreen && selectedTab !== DV_TAB_CHAT_BOARD ? -12 : 0 },
            ]}
            onPress={onPress}
        >
            <View style={localStyles.upperContainer} />
            <View style={localStyles.bottomContainer}>
                <Icon name="arrow-left" size={24} color={colors.Text03} />
            </View>
        </TouchableOpacity>
    )
}

const localStyles = StyleSheet.create({
    upperContainer: {
        width: 57,
        height: 32,
        backgroundColor: 'white',
    },
    bottomContainer: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        width: 57,
        height: 32,
        backgroundColor: 'white',
        borderRightWidth: 1,
        borderRightColor: colors.Gray300,
    },
    touchableContainer: {
        position: 'absolute',
        zIndex: 100,
        width: 57,
        height: 64,
    },
})
