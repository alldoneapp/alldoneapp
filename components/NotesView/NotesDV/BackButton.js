import React from 'react'
import { StyleSheet, View } from 'react-native'
import { useDispatch } from 'react-redux'

import Icon from '../../Icon'
import { colors } from '../../styles/global'
import { TouchableOpacity } from 'react-native-gesture-handler'
import NavigationService from '../../../utils/NavigationService'
import { hideFloatPopup, setSelectedSidebarTab } from '../../../redux/actions'
import store from '../../../redux/store'
import { MIN_URLS_IN_HISTORY } from '../../../URLSystem/URLTrigger'
import { DV_TAB_ROOT_NOTES } from '../../../utils/TabNavigationConstants'
import SharedHelper from '../../../utils/SharedHelper'

export default function BackButton({ parentStyle, upperStyle, buttonStyle, projectId, note }) {
    const dispatch = useDispatch()

    const onPress = () => {
        const { lastVisitedScreen } = store.getState()
        const commonPath = `projects/${projectId}/notes/${note.id}`

        dispatch(hideFloatPopup())
        if (lastVisitedScreen.length >= MIN_URLS_IN_HISTORY) {
            SharedHelper.onHistoryPop(commonPath)
        } else {
            NavigationService.navigate('Root')
            dispatch(setSelectedSidebarTab(DV_TAB_ROOT_NOTES))
        }
    }

    return (
        <TouchableOpacity style={[localStyles.touchableContainer, parentStyle]} onPress={onPress}>
            <View style={[localStyles.upperContainer, upperStyle]} />
            <View style={[localStyles.bottomContainer, buttonStyle]}>
                <Icon name="arrow-left" size={24} color={colors.Text03} />
            </View>
        </TouchableOpacity>
    )
}

const localStyles = StyleSheet.create({
    upperContainer: {
        width: 57,
        height: 32,
        backgroundColor: 'white',
    },
    bottomContainer: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        width: 57,
        height: 32,
        backgroundColor: 'white',
        borderRightWidth: 1,
        borderRightColor: colors.Gray300,
    },
    touchableContainer: {
        width: 57,
        height: 64,
    },
})
