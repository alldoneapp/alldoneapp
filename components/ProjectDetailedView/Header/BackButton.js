import React from 'react'
import { StyleSheet, View, TouchableOpacity } from 'react-native'
import { useDispatch, useSelector } from 'react-redux'

import Icon from '../../Icon'
import { colors } from '../../styles/global'
import NavigationService from '../../../utils/NavigationService'
import { hideFloatPopup, navigateToSettings } from '../../../redux/actions'
import { MIN_URLS_IN_HISTORY } from '../../../URLSystem/URLTrigger'
import { DV_TAB_SETTINGS_PROJECTS } from '../../../utils/TabNavigationConstants'
import SharedHelper from '../../../utils/SharedHelper'

export default function BackButton({ project }) {
    const dispatch = useDispatch()
    const lastVisitedScreen = useSelector(state => state.lastVisitedScreen)

    const onPress = () => {
        const projectId = project.id
        const commonPath = `project/${projectId}`

        dispatch(hideFloatPopup())
        if (lastVisitedScreen.length >= MIN_URLS_IN_HISTORY) {
            SharedHelper.onHistoryPop(commonPath)
        } else {
            NavigationService.navigate('SettingsView')
            dispatch(navigateToSettings({ selectedNavItem: DV_TAB_SETTINGS_PROJECTS }))
        }
    }

    return (
        <TouchableOpacity style={localStyles.touchableContainer} onPress={onPress}>
            <View style={localStyles.upperContainer} />
            <View style={localStyles.bottomContainer}>
                <Icon name="arrow-left" size={24} color={colors.Text03} />
            </View>
        </TouchableOpacity>
    )
}

const localStyles = StyleSheet.create({
    upperContainer: {
        width: 57,
        height: 32,
        backgroundColor: 'white',
    },
    bottomContainer: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        width: 57,
        height: 32,
        backgroundColor: 'white',
        borderRightWidth: 1,
        borderRightColor: colors.Gray300,
    },
    touchableContainer: {
        width: 57,
        height: 64,
    },
})
