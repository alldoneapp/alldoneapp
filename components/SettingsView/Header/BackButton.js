import React from 'react'
import { StyleSheet, View, TouchableOpacity } from 'react-native'
import { useDispatch } from 'react-redux'

import Icon from '../../Icon'
import { colors } from '../../styles/global'
import NavigationService from '../../../utils/NavigationService'
import { hideFloatPopup, navigateToAllProjectsTasks, resetFloatPopup } from '../../../redux/actions'
import store from '../../../redux/store'
import { MIN_URLS_IN_HISTORY } from '../../../URLSystem/URLTrigger'
import SharedHelper from '../../../utils/SharedHelper'

export default function BackButton() {
    const dispatch = useDispatch()

    const onPress = () => {
        const { lastVisitedScreen } = store.getState()

        dispatch(hideFloatPopup())
        if (lastVisitedScreen.length >= MIN_URLS_IN_HISTORY) {
            SharedHelper.onHistoryPop('settings')
        } else {
            NavigationService.navigate('Root')
            dispatch([resetFloatPopup(), navigateToAllProjectsTasks()])
        }
    }

    return (
        <TouchableOpacity style={localStyles.touchableContainer} onPress={onPress}>
            <View style={localStyles.upperContainer} />
            <View style={localStyles.bottomContainer}>
                <Icon name="arrow-left" size={24} color={colors.Text03} />
            </View>
        </TouchableOpacity>
    )
}

const localStyles = StyleSheet.create({
    upperContainer: {
        width: 57,
        height: 32,
        backgroundColor: 'white',
    },
    bottomContainer: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        width: 57,
        height: 32,
        backgroundColor: 'white',
        borderRightWidth: 1,
        borderRightColor: colors.Gray300,
    },
    touchableContainer: {
        width: 57,
        height: 64,
    },
})
