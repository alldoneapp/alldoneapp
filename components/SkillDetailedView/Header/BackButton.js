import React from 'react'
import { StyleSheet, TouchableOpacity } from 'react-native'
import { useSelector, useDispatch } from 'react-redux'

import Icon from '../../Icon'
import { SIDEBAR_MENU_COLLAPSED_WIDTH, colors } from '../../styles/global'
import NavigationService from '../../../utils/NavigationService'
import { hideFloatPopup, setSelectedSidebarTab } from '../../../redux/actions'
import { MIN_URLS_IN_HISTORY } from '../../../URLSystem/URLTrigger'
import { DV_TAB_ROOT_TASKS } from '../../../utils/TabNavigationConstants'
import { getDvLink } from '../../../utils/LinkingHelper'
import SharedHelper from '../../../utils/SharedHelper'

export default function BackButton({ projectId }) {
    const dispatch = useDispatch()
    const lastVisitedScreen = useSelector(state => state.lastVisitedScreen)
    const isMiddleScreen = useSelector(state => state.isMiddleScreen)
    const skillId = useSelector(state => state.skillInDv.id)
    const sidebarExpanded = useSelector(state => state.loggedUser.sidebarExpanded)

    const onPress = () => {
        const commonPath = getDvLink(projectId, skillId, 'skills')

        dispatch(hideFloatPopup())
        if (lastVisitedScreen.length >= MIN_URLS_IN_HISTORY) {
            SharedHelper.onHistoryPop(commonPath)
        } else {
            NavigationService.navigate('Root')
            dispatch(setSelectedSidebarTab(DV_TAB_ROOT_TASKS))
        }
    }

    return (
        <TouchableOpacity
            style={[
                localStyles.container,
                isMiddleScreen ? localStyles.mobileContainer : localStyles.desktopContainer,
                !isMiddleScreen && !sidebarExpanded && { marginLeft: SIDEBAR_MENU_COLLAPSED_WIDTH },
            ]}
            onPress={onPress}
        >
            <Icon name="arrow-left" size={24} color={colors.Text03} />
        </TouchableOpacity>
    )
}

const localStyles = StyleSheet.create({
    container: {
        marginTop: 32,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        width: 57,
        height: 32,
        backgroundColor: 'white',
        borderRightWidth: 1,
        borderRightColor: colors.Gray300,
    },
    mobileContainer: {
        left: -16,
    },
    desktopContainer: {
        position: 'absolute',
        top: 0,
        left: 32,
        zIndex: 100,
    },
})
