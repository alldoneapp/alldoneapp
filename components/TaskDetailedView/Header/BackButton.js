import React from 'react'
import { StyleSheet, View, TouchableOpacity } from 'react-native'
import { useDispatch } from 'react-redux'

import Icon from '../../Icon'
import { colors } from '../../styles/global'
import NavigationService from '../../../utils/NavigationService'
import { hideFloatPopup, setSelectedSidebarTab } from '../../../redux/actions'
import store from '../../../redux/store'
import { MIN_URLS_IN_HISTORY } from '../../../URLSystem/URLTrigger'
import { DV_TAB_ROOT_TASKS } from '../../../utils/TabNavigationConstants'
import { getDvLink } from '../../../utils/LinkingHelper'
import SharedHelper from '../../../utils/SharedHelper'

export default function BackButton({ projectId, task }) {
    const dispatch = useDispatch()

    const onPress = () => {
        const { lastVisitedScreen } = store.getState()
        const commonPath = getDvLink(projectId, task.id, 'tasks')

        dispatch(hideFloatPopup())
        if (lastVisitedScreen.length >= MIN_URLS_IN_HISTORY) {
            SharedHelper.onHistoryPop(commonPath)
        } else {
            NavigationService.navigate('Root')
            dispatch([hideFloatPopup(), setSelectedSidebarTab(DV_TAB_ROOT_TASKS)])
        }
    }

    return (
        <TouchableOpacity style={localStyles.touchableContainer} onPress={onPress}>
            <View style={localStyles.upperContainer} />
            <View style={localStyles.bottomContainer}>
                <Icon name="arrow-left" size={24} color={colors.Text03} />
            </View>
        </TouchableOpacity>
    )
}

const localStyles = StyleSheet.create({
    upperContainer: {
        width: 57,
        height: 32,
        backgroundColor: 'white',
    },
    bottomContainer: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        width: 57,
        height: 32,
        backgroundColor: 'white',
        borderRightWidth: 1,
        borderRightColor: colors.Gray300,
    },
    touchableContainer: {
        width: 57,
        height: 64,
    },
})
