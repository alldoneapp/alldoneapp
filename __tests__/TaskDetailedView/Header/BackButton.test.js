import React from 'react'
import BackButton from '../../../components/TaskDetailedView/Header/BackButton'

import renderer from 'react-test-renderer'

jest.mock('firebase', () => ({ firestore: {} }));

describe('BackButton component', () => {
    describe('BackButton snapshot test', () => {
        it('should render correctly', () => {
            const tree = renderer.create(<BackButton />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
