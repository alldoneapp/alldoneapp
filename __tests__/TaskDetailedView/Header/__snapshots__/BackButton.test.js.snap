// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component BackButton snapshot test should render correctly 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    style={
      Object {
        "height": 64,
        "opacity": 1,
        "width": 57,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "height": 32,
          "width": 57,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRightColor": "#E7ECEF",
          "borderRightWidth": 1,
          "flexDirection": "column",
          "height": 32,
          "justifyContent": "center",
          "width": 57,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#8A94A6",
              "fontFamily": "alldone",
              "fontSize": 24,
              "opacity": 1,
            }
          }
        >
          Ó§ê
        </Text>
      </View>
    </View>
  </View>
</RNGestureHandlerButton>
`;
